### PR TITLE
Update roles and permissions according to newest table

### LINKED ISSUE #5634: ✨ Update seeded Roles & Permissions
## ✨ Feature

Update rolepermission.php so that roles and permissions created match the current [permissions table](https://docs.google.com/spreadsheets/d/1RC7wufxwjLEvaEunHsfZfOzBefg218dvvzkDwKSiQjI/edit?usp=sharing), which has been updated since creating the rolepermission seeder config.

## 🕵️ Details

Mostly, the Team Admin role has been split in two - one role for creating an running recruitments (pools) in a Team, and another role (not team-based) for responding to _all_ search requests.

As always, the names are up for improvement!

Permissions dealing with the administration of a Team, like changing metadata and assigning new members, now belong solely to Super Admins.

## 🎨 Design File

Include a link to the design file (if it exists).

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] After seeding, the roles and permissions match the current [permissions table](https://docs.google.com/spreadsheets/d/1RC7wufxwjLEvaEunHsfZfOzBefg218dvvzkDwKSiQjI/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by tristan-orourke**
🤖 Resolves #5634

## 👋 Introduction

Updates the seeded roles and permissions to match the newest version of [this table](https://docs.google.com/spreadsheets/d/1RC7wufxwjLEvaEunHsfZfOzBefg218dvvzkDwKSiQjI/edit#gid=428263758).

## 🕵️ Details

Add any additional details that could assist with reviewing or testing this PR.

## 🧪 Testing

Not sure what the best way to test this is. Look through the seeder and try to check that nothing is missing compared to the spreadsheet?

Also, try to check that the seeder will run without errors even if you have run the previous version of this seeder.

## 📸 Screenshot

Add a screenshot (if possible).

### CODE REVIEW

**Comment by esizer** [comment]
For the applicant role, there is a check mark in row 53 which has no associated permission. My assumption was this was a mistake in the spreadsheet and should be deleted but I wanted to confirm that.

File: api/config/rolepermission.php [hunk 1]
```diff
@@ -501,11 +545,22 @@
             ],
```
Comment (esizer) [inline api/config/rolepermission.php:545]: The `update-team-team` permission is no longer associated with this role according to the spreadsheet.

**Review by esizer — CHANGES_REQUESTED** [review]
Looking good! I just noticed one thing that does not line up with the excel and the tests will need to be updated with the new roles as well 👍

**Review by esizer — COMMENTED** [review]
Sweet! 🍬

**Review by esizer — APPROVED** [review]

**Comment by tristan-orourke** [comment]
Thanks for the careful check @esizer!

**Pull request merged**